server/applicationset/applicationset.go

```diff
@@ -213,6 +255,27 @@ func (s *Server) Create(ctx context.Context, q *applicationset.ApplicationSetCre
 	return updated, nil
 }
 
+func (s *Server) generateApplicationSetApps(ctx context.Context, appset v1alpha1.ApplicationSet, namespace string) ([]v1alpha1.Application, error) {
+	logCtx := log.WithField("applicationset", appset.Name)
+	gitSubmoduleEnabled := true
+	enableNewGitFileGlobbing := true
+	scmRootCAPath := ""
+	enableScmProviders := true
+	allowedScmProviders := []string{}
```

Comment:
Probably was already on your mind, but we should probably pull these from some common config store. argocd-cmd-params-cm would probably be fine, we just need to make sure folks know that if they're configuring these fields directly on the appset controller deployment, they should switch to using the configmap.

Reply:
Good catch. I forgot I had these to update after the code started to compile again 😬 

My idea was to add them to the config map, set env variable with it in the server, but **not** add parameters to the argocd-server cli, and use these defaults if they are not set.

Reply:
Unfortunately, for the sake of consistency, I think it's worth polluting the API server's CLI interface. If the API server is going to effectively re-implement a chunk of the appset controller, its interface should accept the same parameters. I think it's just a side-effect of expanding the scope of the API server's responsibilities.

I could definitely imagine moving all this config into a shared ConfigMap (besides argocd-cmd-params-cm) at some point and loading them directly from k8s instead of using env vars. But I think that's outside the scope of this PR.

Reply:
I added the parameters to the api server